contracts/messagebuscontract/MessageBus.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./IMessageBus.sol";
+import "./Structs.sol";
+
+import "../libs/openzeppelin/contracts/access/Ownable.sol";
+
+contract MessageBus is IMessageBus, Ownable {
+
+    function messageFee() virtual internal returns (uint256) { return 0; }
+
+    event LogMessagePublished
```

Comment:
this is part of the interface as well right?
basically, it is required that the public function emits it.
(similar to the erc20 transfer event)